operator/controllers/lifecycle/keptnevaluation/controller.go

```diff
@@ -264,10 +265,21 @@ func (r *KeptnEvaluationReconciler) fetchDefinitionAndProvider(ctx context.Conte
 	if err := r.Client.Get(ctx, namespacedDefinition, evaluationDefinition); err != nil {
 		return nil, nil, err
 	}
+
+	if evaluationDefinition.Spec.Source == providers.KeptnMetricProviderName {
+		return evaluationDefinition, &klcv1alpha2.KeptnEvaluationProvider{
```

Comment:
I would pre-compute this `klcv1alpha2.KeptnEvaluationProvider` so we don't need to allocate the same object over and over. This would solve the problem of a possible wrong ns passed to the struct